backend/src/modules/group/groupRepository.ts

```diff
@@ -0,0 +1,174 @@
+import { Prisma } from '@prisma/client';
+import prisma from '../../../prisma/prisma-client';
+import {createGroupInput, Group} from "./groupModel";
+import CustomError from '../../utils/errorModel';
+
+// eslint-disable-next-line import/prefer-default-export
+export const groupRepository = {
+    async createGroup(groupInput: createGroupInput) : Promise<Group> {
+        try {
+            const group = await prisma.group.create({
+                data: {
+                    name: groupInput.name.toLowerCase(),
+                    area: groupInput.area || '',
+                },
+            });
+
+            return {
+                id: group.id,
+                name: group.name,
+                area: group.area ?? '',
+                createdAt: group.createdAt,
+                updatedAt: group.updatedAt ?? new Date(),
+            };
+        } catch (error) {
+            if (error instanceof Prisma.PrismaClientKnownRequestError) {
+                if (error.code === 'P2002') {
+                    throw new CustomError(409, 'Role already exists');
```

Comment:
"Role alreay exists" ??? when creating a group? hmmmmm. Should be "Group already exists"

Reply:
It should be Group already exists, this is a copy and paste error.